azurerm/internal/services/iotcentral/tests/resource_arm_iotcentral_application_test.go

```diff
@@ -0,0 +1,232 @@
+	data := acceptance.BuildTestData(t, "azurerm_iotcentral_application", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMIotCentralApplicationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMIotCentralApplication_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMIotCentralApplicationExists(data.ResourceName),
+				),
+			},
+			{
+				Config:      testAccAzureRMIotCentralApplication_requiresImport(data),
+				ExpectError: acceptance.RequiresImportError("azurerm_iotcentral_application"),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMIotCentralApplicationExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found:  %s", resourceName)
+		}
+		appName := rs.Primary.Attributes["name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for IoT Central Application:  %s", appName)
+		}
+
+		client := acceptance.AzureProvider.Meta().(*clients.Client).IoTCentral.AppsClient
+		resp, err := client.Get(ctx, resourceGroup, appName)
+		if err != nil {
+			if resp.StatusCode == http.StatusNotFound {
+				return fmt.Errorf("Bad: IoT Central Application %q (Resource Group  %q) does not exist", appName, resourceGroup)
+			}
+
+			return fmt.Errorf("Bad: Get IoT Central Application:  %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMIotCentralApplicationDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).IoTCentral.AppsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_iotcentral_application" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		if resp, err := client.Get(ctx, resourceGroup, name); err != nil {
+			if resp.StatusCode != http.StatusNotFound {
+				return fmt.Errorf("Bad: Get IoT Central Application:  %+v", err)
+			}
+		}
+	}
+	return nil
+}
+
+func testAccAzureRMIotCentralApplication_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+        resource "azurerm_resource_group" "test" {
+           name     = "acctestRG_%[2]d"
+           location = "%[1]s"
+        }
+		
+        resource "azurerm_iotcentral_application" "test" {
+          name                = "acctestiotcentralapp-%[2]d"
+          location            = "${azurerm_resource_group.test.location}"
+          resource_group_name = "${azurerm_resource_group.test.name}"
+          sub_domain          = "acctestiotcentralapp-%[2]d"
```

Comment:
acctest-IoTCentralApp-%[2]d, right? Please also update the same issue in other place.